interfaces/builtin/desktop.go

```diff
@@ -609,9 +616,37 @@ func (iface *desktopInterface) AppArmorPermanentSlot(spec *apparmor.Specificatio
 	return nil
 }
 
+func (iface *desktopInterface) validateDesktopFileIDs(attribs interfaces.Attrer) error {
+	attrVal, exists := attribs.Lookup("desktop-file-ids")
+	if !exists {
+		// desktop-file-ids attribute is optional
+		return nil
+	}
+
+	// desktop-file-ids must be a list of strings.
+	desktopFileIDs, ok := attrVal.([]interface{})
+	if !ok {
+		return fmt.Errorf(`cannot add %s plug: "desktop-file-ids" must be a list of strings`, iface.name)
+	}
+	for _, desktopFileID := range desktopFileIDs {
+		if _, ok := desktopFileID.(string); !ok {
+			return fmt.Errorf(`cannot add %s plug: "desktop-file-ids" must be a list of strings`, iface.name)
+		}
+	}
+
```

Comment:
Should `desktopFileIDs` be further validated as per https://specifications.freedesktop.org/desktop-entry-spec/latest/file-naming.html ? Whilst I understand that a snap declaration is required for the use of any entries here, I wonder if we should have more of a defence-in-depth approach and not just rely on the store for this validation but instead have snapd validate these?

Reply:
Great point, Thanks for pointing this out! I updated the validation logic and added a corresponding test.